### ISSUE TITLE
[data grid] When deleting a row on the Data Grid using keyboard, the focus after deleting should go to the next line

### ISSUE THREAD

**Issue by joserodolfofreitas** [mui/mui-x#14993]
## Summary

When we delete one line with the delete button, focus should be moved on the first cell of the next line

_Reported by iValua accessibility report._

## Steps to reproduce
- Go to https://mui.com/x/react-data-grid/editing/#full-featured-crud
- Navigate to the delete action button on a row using the keyboard
- Input enter on keyboard
- The row is deleted but the Data Grid loses focus.

## Caveats

They haven't provided a reproduction case, so I'm guessing the problem is when using action buttons, which is not a direct responsibility of the Data Grid, and if that's the case, how could we solve it?

**Search keywords**:

**Comment by joserodolfofreitas** [comment]
@abdermaiza, could you please confirm the steps to reproduce this? Does it indeed involve using the action button?

**Comment by abdermaiza** [comment]
I confirm, we just need to move the focus on the next line, after deleting the line.

**Comment by abdermaiza** [comment]
Hello,

Doesn't seem ok, on this URL:
https://deploy-preview-15061--material-ui-x.netlify.app/x/react-data-grid/editing/

The focus is on the edit button instead of the fist cell (Name cell)...

**Comment by cherniavskii** [comment]
@abdermaiza Hmm, do you expect the focus to move to the first cell?
To me, it seems natural that the focus stays in the same column. Moreover, when there are a lot of columns with horizontal scroll, wouldn't it be strange to scroll all the way to the first cell?
What do you think?

**Comment by cherniavskii** [comment]
I checked the [ARIA guide for the Grid pattern](https://www.w3.org/WAI/ARIA/apg/patterns/grid/), but I couldn't find any info regarding focus after deleting a row. There's only a general keyboard navigation block which doesn't clarify it much:

<img width="952" src="https://github.com/user-attachments/assets/b9d2e625-bec1-4e51-84b1-8460d877ee64">

[Source](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#discernibleandpredictablekeyboardfocus)

However, I found this example they provide that uses the same Grid pattern: https://www.w3.org/WAI/ARIA/apg/patterns/grid/examples/layout-grids/#ex2_label.

When the row is deleted, the focus moves to the equivalent cell in the next row:

https://github.com/user-attachments/assets/65702665-017c-424d-96e1-ebafc8b9e2cf

This makes me believe that keeping the focus in the same column is technically correct and should be the default behavior.
Are you sure you want a different behavior? We can consider making it configurable.

**Comment by abdermaiza** [comment]
I agree but this is not what is done in the example here:
https://deploy-preview-15061--material-ui-x.netlify.app/x/react-data-grid/editing/

Because I see that the focus is moved in the edit button of the deleted row... :(

**Comment by cherniavskii** [comment]
Well, the focus moves to the cell in the same column, and it automatically focuses the first button in the cell 🙂
The demo from the guide I shared above only has one button in the cell, in which case the behavior would be identical for the Data Grid.

**Comment by abdermaiza** [comment]
Ok, so that's fine, thanks.
